app/lib/dartdoc/backend.dart

```diff
@@ -43,6 +46,60 @@ class DartdocBackend {
 
   DartdocBackend(this._db, this._storage);
 
+  /// Whether the storage bucket has a useable extracted data file.
+  /// Only the existence of the file is checked.
+  // TODO: decide whether we should re-generate the file after a certain age
+  Future<bool> hasValidDartSdkDartdocData() async {
+    final objectName =
+        storage_path.dartSdkDartdocDataName(shared_versions.runtimeVersion);
+    try {
+      final info = await _storage.info(objectName);
+      return info != null;
+    } catch (_) {
+      return false;
+    }
+  }
+
+  /// Upload the generated dartdoc data file for the Dart SDK to the storage bucket.
+  Future uploadDartSdkDartdocData(File file) async {
+    final objectName =
+        storage_path.dartSdkDartdocDataName(shared_versions.runtimeVersion);
+    try {
+      final contentBytes = await file.readAsBytes();
+      await _storage.writeBytes(objectName, _gzip.encode(contentBytes));
+    } catch (e, st) {
+      _logger.warning(
+          'Unable to upload SDK pub dartdoc data file: $objectName', e, st);
```

Comment:
Is this likely? Should it be an error? Can we catch unlikely issues here? Can we catch a more specific exception?

Reply:
On the larger scale of uploading thousands of files of the dartdoc-generated files, it does happen (request timeout, network issue, an occasional error 500 from the AppEngine service), but on a per-file basis, it may be rare. This specific file is larger then the rest, and it may happen more frequently.
I wouldn't make it a hard error, because at worst case we won't be able to search in the API docs for a limited period, which should be rather small (measured in minutes), as eventually another dartdoc instance would try to generate and upload it too.

Reply:
Alright sounds good